MovieApp/MovieDetailsViewController.swift

```diff
@@ -0,0 +1,351 @@
+            $0.leadingMargin.equalTo(contentView.safeAreaLayoutGuide).inset(30)
+            $0.trailing.equalToSuperview().offset(-30)
+            $0.top.equalTo(descriptionLabel.snp.bottom).offset(20)
+        }
+        stackViewBottom.snp.makeConstraints {
+            $0.leadingMargin.equalTo(contentView.safeAreaLayoutGuide).inset(30)
+            $0.trailing.equalToSuperview().offset(-30)
+            $0.top.equalTo(stackViewTop.snp.bottom).offset(40)
+        }
+    }
+    
+    private func setStyles() {
+        view.backgroundColor = .white
+        
+        scoreLabel.text = "76%"
+        scoreLabel.font = .systemFont(ofSize: 12, weight: .bold)
+        scoreLabel.textColor = .white
+        userScoreLabel.text = "User Score"
+        userScoreLabel.font = .systemFont(ofSize: 12, weight: .bold)
+        userScoreLabel.textColor = .white
+        
+        titleLabel.text = "Iron Man"
+        titleLabel.font = .systemFont(ofSize: 24, weight: .bold)
+        titleLabel.textColor = .white
+        yearLabel.text = "(2008)"
+        yearLabel.font = .systemFont(ofSize: 24)
+        yearLabel.textColor = .white
+        
+        dateLabel.text = "05/02/2008"
+        dateLabel.font = .systemFont(ofSize: 12)
+        dateLabel.textColor = .white
+        countryLabel.text = "(US)"
+        countryLabel.font = .systemFont(ofSize: 12)
+        countryLabel.textColor = .white
+        
+        genreLabel.text = "Action, Science Fiction, Adventure"
+        genreLabel.font = .systemFont(ofSize: 12)
+        genreLabel.textColor = .white
+        durationLabel.text = "2h 6m"
+        durationLabel.font = .systemFont(ofSize: 12, weight: .bold)
+        durationLabel.textColor = .white
+        
+        starGraphicView.layer.backgroundColor = UIColor(red: 0.5, green: 0.5, blue: 0.5, alpha: 0.6).cgColor
+        starGraphicView.layer.cornerRadius = 20
+        starGraphicView.clipsToBounds = false
+        starGraphicView.tintColor = .white
+        
+        
+        overviewLabel.text = "Overview"
+        overviewLabel.font = .systemFont(ofSize: 24, weight: .bold)
+        overviewLabel.textColor = .black
+        
+        descriptionLabel.text = "After being held captive in an Afghan cave, billionaire engineer Tony Stark creates a unique weaponized suit of armor to fight evil."
+        descriptionLabel.font = .systemFont(ofSize: 12)
+        descriptionLabel.numberOfLines = 0
+    }
+}
+
+extension MovieDetailsViewController: UIScrollViewDelegate {
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        print("scrollViewDidScroll")
+    }
+
+    func scrollViewDidScrollToTop(_ scrollView: UIScrollView) {
+        print("scrollViewDidScrollToTop")
+    }
+
+    func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
+        print("scrollViewDidEndDragging")
+    }
+}
+
+class CustomRoleView:UIView {
+    private var name: String?
+    private var role: String?
```

Comment:
These properties don't have to be optional as this view will always have a name and a role. Also, I don't see these properties changed anywhere, so they could probably be a `let` instead of `var` .